libs/proposals/src/components/protocol-upgrade-in-progress-notification.tsx

```diff
@@ -0,0 +1,69 @@
+import { useProtocolUpgradeProposalLink } from '@vegaprotocol/environment';
+
+import { t } from '@vegaprotocol/i18n';
+import {
+  ExternalLink,
+  Intent,
+  NotificationBanner,
+  SHORT,
+} from '@vegaprotocol/ui-toolkit';
+import type { StoredNextProtocolUpgradeData } from '../lib';
+import {
+  NEXT_PROTOCOL_UPGRADE_PROPOSAL_SNAPSHOT,
+  useNextProtocolUpgradeProposal,
+} from '../lib';
+import { useLocalStorageSnapshot } from '@vegaprotocol/react-helpers';
+import { useBlockRising } from '../lib/protocol-upgrade-proposals/use-block-rising';
+
+/**
+ * A flag determining whether to get the upgrade proposal data from local
+ * storage if `useNextProtocolUpgradeProposal` fails.
+ */
+const ALLOW_STORED_PROPOSAL_DATA = true;
+
+export const ProtocolUpgradeInProgressNotification = () => {
+  const { data, error } = useNextProtocolUpgradeProposal(undefined, true);
+  const [nextUpgrade] = useLocalStorageSnapshot(
+    NEXT_PROTOCOL_UPGRADE_PROPOSAL_SNAPSHOT
+  );
+  const { blocksRising, block } = useBlockRising();
+  const detailsLink = useProtocolUpgradeProposalLink();
+
+  const { vegaReleaseTag, upgradeBlockHeight } =
+    error && !data && nextUpgrade && ALLOW_STORED_PROPOSAL_DATA
+      ? (JSON.parse(nextUpgrade) as StoredNextProtocolUpgradeData) // gets stored data in case of an data node error if allowed
+      : data || { vegaReleaseTag: undefined, upgradeBlockHeight: undefined };
+
+  /**
+   * If upgrade is in progress then none of the nodes should produce blocks,
+   * same should be with the tendermint block info otherwise it's a network
+   * issue and not an upgrade. The upgrade usually lasts for couple of minutes.
+   *
+   * Once the networks is back then the notification disappears.
+   */
+  const UPGRADE_IN_PROGRESS =
```

Comment:
Reserve upper camel case for a 'true' constant

```suggestion
  const upgradeInProgress =
```